### PR TITLE
libdrm: add recipe

### PR DESCRIPTION
**Pull request by kenmays**
Tested on Haiku R1B3 x86 and x86_64 platforms.
Supports:
- AMD Radeon HD 2000 series through Radeon RX 6000 series (Radeon RX 6900 XT)
- Intel GPUs up to Gen 12 (Tiger Lake, Rocket Lake, (Alder Lake processors))
- Nvidia Riva TNT through GeForce GTX 1070 (GeForce RTX 2060 (W-I-P))

### CODE REVIEW

File: sys-libs/libdrm/libdrm-2.4.109.recipe [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		2009-2015 VMware, Inc.
+		2008, 2013 Red Hat
+		2014-2015 Broadcom
+		2014 Advanced Micro Devices, Inc.
+		"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://dri.freedesktop.org/libdrm/libdrm-${portVersion}.tar.xz"
+CHECKSUM_SHA256="629352e08c1fe84862ca046598d8a08ce14d26ab25ee1f4704f993d074cb7f26"
+SOURCE_DIR="libdrm-${portVersion}"
```
Comment (extrowerk) [inline sys-libs/libdrm/libdrm-2.4.109.recipe:36]: necessary?

File: sys-libs/libdrm/libdrm-2.4.109.recipe [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+libVersion=2.4.0
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	libdrm$secondaryArchSuffix = $portVersion
+	lib:libdrm$secondaryArchSuffix = $libVersionCompat
+	lib:libdrm_amdgpu$secondaryArchSuffix = 1.0.0 
+	lib:libdrm_intel$secondaryArchSuffix = 1.0.0
+	lib:libdrm_nouveau$secondaryArchSuffix = 2.0.0
```
Comment (extrowerk) [inline sys-libs/libdrm/libdrm-2.4.109.recipe:50]: but the meson invocation says: "-Dnouveau=false -Dintel=false "

File: sys-libs/libdrm/libdrm-2.4.109.recipe [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+INSTALL()
+{
+	ninja -C build install
+
+	# Do a little cleanup
+	rm -rf $prefix/share
+
+	# Set some nice version info
+#	appVersion=${portVersion//\./ }
```
Comment (extrowerk) [inline sys-libs/libdrm/libdrm-2.4.109.recipe:112]: remove if unneeded

**Review by waddlesplash — CHANGES_REQUESTED** [review]
As far as I'm aware, @X547's work uses a totally separate implementation of libdrm, not this one.

**Comment by X547** [comment]
I use original libdrm_amdgpu and libdrm headers, but libdrm.so code is different.

**Comment by kenmays** [comment]
Using libdrm2 framework instead.

**Pull request closed without merging**